docs/RELEASE_PROCESS.md

```diff
@@ -0,0 +1,227 @@
+# Rally Release Process
+The Rally Core Add-on will be shipped as a [browser extension](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions), with special Firefox-only privileges.
+In order to enable Firefox-specific privileges it requires a signature done on Mozilla infrastructure, in the [`mozilla-extensions`](https://github.com/mozilla-extensions/ion-core-addon/) clone of [this repository](https://github.com/mozilla-ion/ion-core-addon).
+
+The development & release process roughly follows the [GitFlow model](https://nvie.com/posts/a-successful-git-branching-model/).
+
+> **Note:** The rest of this section assumes that `upstream` points to the `https://github.com/mozilla-ion/ion-core-addon` repository,
+> `origin` points to the developer fork and `mozext` refers to `https://github.com/mozilla-extensions/ion-core-addon`.
+> For some developer workflows, `upstream` can be the same as `origin`.
+
+**Table of Contents**:
+
+* [Published artifacts](#published-artifacts)
+* [Standard release](#standard-release)
+* [Hotfix release for latest version](#hotfix-release-for-latest-version)
+* [Hotfix release for previous version](#hotfix-release-for-previous-version)
+
+## Published artifacts
+
+* A QA-signed version of the Firefox Rally Core Add-on (**only meant for QA**).
+* A Mozilla-signed version of the Firefox Rally Core Add-on (**TODO**).
```

Comment:
I filed https://github.com/mozilla-ion/ion-core-addon/issues/242 for the TODO here.